src/test/java/harness/utils/CleanupTestUserWorkspaces.java

```diff
@@ -0,0 +1,115 @@
+    testUser.login();
+
+    // `terra workspace list`
+    List<UFWorkspace> listWorkspaces =
+        TestCommand.runAndParseCommandExpectSuccess(new TypeReference<>() {}, "workspace", "list");
+
+    List<UFWorkspaceUser> listWorkspaceUsers;
+    for (UFWorkspace workspace : listWorkspaces) {
+      try {
+        // `terra workspace list-users`
+        listWorkspaceUsers =
+            TestCommand.runAndParseCommandExpectSuccess(
+                new TypeReference<>() {}, "workspace", "list-users", "--workspace=" + workspace.id);
+
+        // find the user in the list
+        Optional<UFWorkspaceUser> workspaceUser =
+            listWorkspaceUsers.stream()
+                .filter(user -> user.email.equalsIgnoreCase(testUser.email))
+                .findAny();
+        // skip deleting if the test user is not an owner
+        if (workspaceUser.isEmpty() || !workspaceUser.get().roles.contains(IamRole.OWNER)) {
+          System.out.println(
+              "Skip deleting workspace because test user is not an owner: id="
+                  + workspace.id
+                  + ", testuser="
+                  + testUser.email);
+          continue;
+        }
+
+        System.out.println(
+            "Deleting workspace: id=" + workspace.id + ", testuser=" + testUser.email);
+        if (!isDryRun) {
+          // `terra workspace delete --workspace=$id`
+          TestCommand.runCommandExpectSuccess("workspace", "delete", "--workspace=" + workspace.id);
+          System.out.println(
+              "Cleaned up workspace: id=" + workspace.id + ", testuser=" + testUser.email);
+        }
+      } catch (Throwable ex) {
+        System.out.println(
+            "Error deleting workspace: id=" + workspace.id + ", testuser=" + testUser.email);
+        continue;
+      }
+    }
+
+    // `terra auth revoke`
+    TestCommand.runCommandExpectSuccess("auth", "revoke");
+  }
+
+  /**
+   * Loop through all test users with spend profile access and try to delete all workspaces they
+   * own.
+   */
+  public static void main(String... args) {
+    // (see the Gradle cleanupTestUserWorkspaces task for how these System properties get set from
+    // Gradle properties)
+    String server = System.getProperty("TERRA_SERVER");
+    if (server == null || server.isEmpty()) {
+      throw new UserActionableException("Specify the server to cleanup.");
+    }
+    TestCommand.runCommandExpectSuccess("server", "set", "--name", server);
+
+    boolean isDryRun = Boolean.parseBoolean(System.getProperty("DRY_RUN"));
+
+    System.out.println("TERRA_SERVER: " + server);
+    System.out.println("DRY_RUN: " + isDryRun);
+
+    Arrays.stream(TestUsers.values())
+        .filter(TestUsers::hasSpendAccess)
+        .forEach(
+            testUser -> {
+              try {
+                deleteWorkspaces(testUser, isDryRun);
+              } catch (IOException e) {
+                System.out.println("Error cleaning up workspaces for testuser: " + testUser.email);
```

Comment:
Is it possible to list the workspaces that failed to be deleted? You'd have to keep a running list.

Reply:
Sure. Added a summary printout at the end.